src/rhsmlib/dbus/objects/register.py

```diff
@@ -152,19 +170,27 @@ def _register(self, org, activation_keys, options):
         return registration_output
 
     def build_uep(self, options):
-        return connection.UEPConnection(
-            username=options.get('username', None),
-            password=options.get('password', None),
-            host=options.get('host', None),
-            ssl_port=connection.safe_int(options.get('port', None)),
-            handler=options.get('handler', None),
-            insecure=options.get('insecure', None),
-            proxy_hostname=options.get('proxy_hostname', None),
-            proxy_port=options.get('proxy_port', None),
-            proxy_user=options.get('proxy_user', None),
-            proxy_password=options.get('proxy_password', None),
-            restlib_class=connection.BaseRestLib
-        )
+        connection_info = {}
+        server_sec = conf['server']
+        connection_info['host'] = options.setdefault('host', server_sec['hostname'])
+        connection_info['ssl_port'] = options.setdefault('port', server_sec.get_int('port'))
+        connection_info['handler'] = options.setdefault('handler', server_sec['prefix'])
+        connection_info['proxy_hostname_arg'] = options.setdefault('proxy_hostname', server_sec['proxy_hostname'])
+        connection_info['proxy_port_arg'] = options.setdefault('proxy_port', server_sec.get_int('proxy_port'))
+        connection_info['proxy_user_arg'] = options.setdefault('proxy_user', server_sec['proxy_user'])
+        connection_info['proxy_password_arg'] = options.setdefault('proxy_password', server_sec['proxy_password'])
+        connection_info['no_proxy_arg'] = options.setdefault('no_proxy', server_sec['no_proxy'])
```

Comment:
Yeah, not sure why I used `setdefault` there instead of just `get`.  Should be corrected now.